src/as_execution/mod.rs

```diff
@@ -128,9 +128,9 @@ pub(crate) fn init_sp_engine(limit: u64, gas_costs: GasCosts) -> Engine {
         compiler_config.push_middleware(metering);
     }
 
-    EngineBuilder::new(compiler_config)
+    Engine::from(EngineBuilder::new(compiler_config)
```

Comment:
don't know if it's a good way; 

`EngineBuilder::new(config).engine()` return `wasmer_compiler::engine::inner::Engine`
and previous return type wanted is `wasmer::engine::Engine`